### DISCUSSION TITLE
Is it possible change AI Chat Title?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#14684] [Q&A]
Hi,
    the default ai-chat title is Chat [EXPERIMENTAL], which is not good enough for product, is it possible to change it by custom extension code?
![image](https://github.com/user-attachments/assets/0881f0c0-4c36-4938-9e12-9cb469dacaf5)

**Comment by planger** [comment]
Thank you for raising this issue!

While it should still be possible to overwrite the name of the `ChatViewWidget` by rebinding it to a custom subclass that overwrites the constructor and sets a custom name (see below), it indeed is very inconvenient, as almost any product that uses it would need to do that.
Especially since we are planning to move Theia AI out of beta state soon.

I've created [an issue](https://github.com/eclipse-theia/theia/issues/14687) to improve the situation!

As a workaround it should work to rebind `ChatViewWidget`:

```ts
container.rebind(ChatViewWidget).to(CustomChatViewWidget);
```

And overwrite the constructor in your custom subclass:

```ts
@injectable()
export class CustomChatViewWidget extends ChatViewWidget {
    constructor(
        @inject(ChatViewTreeWidget) treeWidget: ChatViewTreeWidget,
        @inject(AIChatInputWidget) inputWidget: AIChatInputWidget
    ) {
        super(treeWidget, inputWidget);
        // Override the label
        this.title.label = `Custom Label`; // New label
        this.title.caption = `Custom Label`; // Optional: for the tooltip
    }
}
```

Be aware that the title is also set in teh `AIChatContribution`, directly accessing the static field in `ChatViewWidget.LABEL`. So you would probably also have to remove that contribution and overwrite it. Here a [contribution filter](https://theia-ide.org/docs/contribution_filter/) may help.

Note that I haven't tested the workaround above directly. So I'm not 100 % sure that the code is correct and works.

**Reply by wss29** [reply]
hi @planger 
   I have indeed considered creating a custom widget to rebind the ChatViewWidget, but I feel it’s not worth making such significant changes just to modify a title. Therefore, I’m looking forward to a new method in a future release. Thanks again!
